src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/SocketOutput.cs

```diff
@@ -192,6 +192,10 @@ public class SocketOutput : ISocketOutput
                 {
                     _postingWrite = true;
                     _ongoingWrites++;
+                    // We don't know how long it will take to complete the write, but we don't want to
+                    // close the connection knowing there is a pending/ongoing write, since we know there will
+                    // be connection activity. Hence, disable the timeout until there are no ongoing writes.
+                    _connection.ConnectionControl.DisableTimeout();
```

Comment:
Because a single write can be arbitrarily long. This is why I re-enable the timeout once all writes are completed.